plugins/modules/dcnm_res_manager.py

```diff
@@ -0,0 +1,1254 @@
+    type: str
+    required: true
+  state:
+    description:
+      - The required state of the configuration after module completion.
+    type: str
+    required: false
+    choices:
+      - merged
+      - deleted
+      - query
+    default: merged
+  config:
+    description:
+      - A list of dictionaries containing resources and switch information
+    type: list
+    elements: dict
+    suboptions:
+      entity_name:
+        description:
+          - A unique name which identifies the entity to which the resourcce is allocated to.
+          - The format of this parameter depends on the scope_type. The details are provided in
+          - the EXAMPLES section
+        type: str
+        required: true
+      pool_type:
+        description:
+          - Type of resource pool
+        type: str
+        required: true
+        choices:
+          - ID
+          - IP
+          - SUBNET
+      pool_name:
+        description:
+          - Name of the resource pool from which the resource is allocated
+        type: str
+        required: true
+      scope_type:
+        description:
+          - Socpe of resource allocation
+        type: str
+        required: true
+        choices:
+          - fabric
+          - device
+          - device_interface
+          - device_pair
+          - link
+      resource:
+        description:
+          - Value of the resource being allocated
+          - The value will be
+          -     an integer if pool_type is ID
+          -     an IPV4/IPV6 address if pool_type is IP
+          -     an IPV4 address/net_mask or IPV6 address/net_maskif pool_type is SUBNET
+        type: str
+        required: true
+      switch:
+        description:
+        - IP address or DNS name of the management interface of the switch to which the allocated resource is assigned to.
+        type: list
+        required: false
+"""
+
+EXAMPLES = """
+# Entity name format
+# ==================
+#
+# The formt of the entity name depends on the scope_type of the resource being allocated.
+
+# Scope Type                Entity Name
+# =====================================
+# Fabric                    Eg: My_Network_30000
+# Device                    Eg: loopback0
+# Device Pair               Eg: FDO21331S8T~FDO21332E6X~vPC1
+# Device Interface          Eg: FDO21332E6X~Ethernet1/13
+# Link                      Eg: FDO21332E6X~Ethernet1/3~FDO21331S8T~Ethernet1/3
+
+# where FDO21331S8T and FDO21331S8T are switch serial numbers
+
+# This module supports the following states:
+
+# Merged:
+#   Resources defined in the playbook will be merged into the target fabric.
+#     - If the Resources does not exist it will be added.
+#     - If the Resources exists but properties managed by the playbook are different
+#       they will be updated if possible.
+#     - Resources that are not specified in the playbook will be untouched.
+#
+# Deleted:
+#   Resources defined in the playbook will be deleted.
```

Comment:
Do we get an error message informing the user if we don't specify a resource with state deleted?   Just want to make sure since we support this in some module but not this one.

Reply:
As discussed, all parameters are mandatory for state DELETED. So if a parameter is not included it will error out.